src/component/toolbox/feature/DataView.js

```diff
@@ -426,13 +432,18 @@ function tryMergeDataOption(newData, originalData) {
     return zrUtil.map(newData, function (newVal, idx) {
         var original = originalData && originalData[idx];
         if (zrUtil.isObject(original) && !zrUtil.isArray(original)) {
-            if (zrUtil.isObject(newVal) && !zrUtil.isArray(newVal)) {
-                newVal = newVal.value;
+            var newValIsObject = zrUtil.isObject(newVal) && !zrUtil.isArray(newVal);
+            if (!newValIsObject) {
+                newVal = {
+                    value: newVal
+                };
+            }
+            // original data has name but new data has no name
+            if (original.name != null && newVal.name == null) {
+                delete original.name;
             }
             // Original data has option
```

Comment:
Should not modify the original data. It's better to delete the name in newVal after `defaults`

Reply:
Yes, I missed it before. I've tweaked this logic, please help review, thank you.